NameLayerMain/src/vg/civcraft/mc/namelayer/database/AssociationList.java

```diff
@@ -61,42 +57,17 @@ public void initializeProcedures(){
 				+ "in pl varchar(40), in uu varchar(40)) sql security invoker begin "
 				+ ""
 				+ "declare account varchar(40);"
-				+ "declare nameamount int(10);"
-				+ ""
-				+ "set @@SESSION.max_sp_recursion_depth = 30;"
-				+ ""
-				+ "set nameamount=0;"
-				+ "set nameamount=(select count(*) from Name_player p where p.uuid=uu);"
-				+ ""
-				+ "if (nameamount < 1) then"
-				+ "		setName: loop"
-				+ "		set account =(select uuid from Name_player p where p.player=pl);"
-				+ "		if (account not like uu) then"
-				+ ""
-				+ "				if (nameamount > 0) then"
-				+ "					set pl = (select concat(SUBSTRING(pl, 1, length(pl)-1)));"
-				+ "				end if;"
-				+ ""
-				+ "			insert ignore into playercountnames (player, amount) values (pl, 0);"
-				+ ""
-				+ "			update playercountnames set amount = nameamount+1 where player=pl;"
-				+ ""
-				+ "			set nameamount=(select amount from playercountnames where player=pl);"
-				+ ""
-				+ "			set pl = (select concat (pl,nameamount));"
+				+ "declare counter int(10);"
 				+ ""
-				+ "			set account =(select uuid from Name_player p where p.player=pl);"
+				+ "set counter=0;"
+				+ "set counter=(select count(*) from Name_player p where p.uuid=uu);" //count of all entries with the players uuid
 				+ ""
-				+ "			if (account not like uu) then"
-				+ "				iterate setName;"
-				+ "			end if;"
-				+ "		else"
-				+ "			insert ignore into Name_player (player, uuid) values (pl, uu);"
-				+ "			leave SetName;"
+				+ "if (counter == 0) then" //this uuid is not in the table yet
```

Comment:
Ah okay, I'll test this later anyway on civtest before throwing it up, to make sure I didnt break the procedure.